Libraries/LibGfx/ImageFormats/BMPLoader.cpp

```diff
@@ -1383,7 +1383,14 @@ static ErrorOr<void> decode_bmp_pixel_data(BMPLoadingContext& context)
                 u8 g = (pixel & 0x00FF00) >> 8;
                 u8 r = (pixel & 0x0000FF);
 
-                u32 rgbx_pixel = (r << 16) | (g << 8) | b;
+                u32 rgbx_pixel;
+                // 24bit ICO files use BGRA8888 format
+                if (context.is_included_in_ico) {
+                    rgbx_pixel = (b << 16) | (g << 8) | r | 0xFF000000;
```

Comment:
Am I understanding right that if `is_included_in_ico`, that `b` isn't actually the blue channel, etc? (Because we're reading BGRA, not writing it.)

If so then it'd be clearer to assign `b`, `g` and `r` to the correct parts of the color data instead.